tests/e2e/volumepropverification_test.go

```diff
@@ -24,118 +24,122 @@ import (
-				expctdPropsAdmin = []string{"100MB", formatTypes[k]}
-				if !hasElement(expctdPropsAdmin, volmPropertiesAdminCli) {
-					log.Fatal("Volume properties on admin cli do not matches with the expected values")
-				}
-				log.Println("Finished verifying volume properties like size, disk-format and attached-to-vm fields"+
-					" at vm and esx for volume - ", volName)
-				ssh.InvokeCommand(vms[vmIndx], "docker stop "+containerName+" ; docker rm "+containerName)
-			}
+
+func (s *VolumePropVerificationSuite) TestVolumeProperties(c *C) {
+	log.Printf("START: TestVolumeProperties")
+	formatTypes := []string{"thin", "zeroedthick", "eagerzeroedthick"}
+	for i := 0; i < len(formatTypes); i++ {
+		s.volumeName = inputparams.GetVolumeNameWithTimeStamp(formatTypes[i])
+		s.containerName = inputparams.GetContainerNameWithTimeStamp("volumepropverification")
+
+		log.Printf("Verifying volume properties like size, disk-format and attached-to-vm fields "+
+			"at vm and esx for volume [%s] ", s.volumeName)
+		out, err := ssh.InvokeCommand(s.dockerHostIP, con.CreateVolume+"--name="+s.volumeName+" -o diskformat="+formatTypes[i])
+		c.Assert(err, IsNil, Commentf(out))
+
+		volumePropsAdminCli := verification.GetVolumePropertiesAdminCli(s.volumeName, s.esxIP)
+		expectedPropsAdminCli := []string{"100MB", formatTypes[i], "detached"}
+		c.Assert(hasElement(expectedPropsAdminCli, volumePropsAdminCli), Equals, true, Commentf("Volume %s properties "+
+			"on ESX fetched using admin cli does not matches with the expected values", s.volumeName))
+
+		volumePropsDockerCli := verification.GetVolumePropertiesDockerCli(s.volumeName, s.dockerHostIP)
+		expectedPropsDockerCli := []string{"100MB", formatTypes[i], "<no value>"}
+		c.Assert(hasElement(expectedPropsDockerCli, volumePropsDockerCli), Equals, true, Commentf("Volume %s properties "+
+			"on docker vm fetched using docker cli does not matches with the expected values", s.volumeName))
+
+		if formatTypes[i] == "thin" {
+			out, err = dockercli.AttachVolume(s.dockerHostIP, s.volumeName, s.containerName)
+			c.Assert(err, IsNil, Commentf("Failed to attach the volume [%s]", s.volumeName))
+
+			// Verifying attached to the vm field for volume
+			vmNameFromAdminCli := verification.GetVMAttachedToVolUsingAdminCli(s.volumeName, s.esxIP)
+			vmNameFromDockerCli := verification.GetVMAttachedToVolUsingDockerCli(s.volumeName, s.dockerHostIP)
+
+			// Get vm name based on ip and compare ith with the docker cli and admin cli
+			c.Assert(vmNameFromAdminCli, Equals, s.dockerHostName, Commentf("VM name fetched from attached-to-VM field "+
+				"for volume %s is not same as that fetched from its IP.", s.volumeName))
+
+			c.Assert(vmNameFromAdminCli, Equals, vmNameFromDockerCli, Commentf("Information mis-match - "+
+				"Attached-to-VM field for volume from docker cli is [%s] and attched-to-vm field from admin cli is [%s]", vmNameFromDockerCli, vmNameFromAdminCli))
```

Comment:
Why shouldn't these two checks on the VM name be done for all disk format types - why is it done only for "thin" disks?